From 508ee9985dd4d8a90fc3695df4643ca82d265cc4 Mon Sep 17 00:00:00 2001
From: Chih-Hung Hsieh <chh@google.com>
Date: Wed, 29 Jun 2016 11:42:46 -0700
Subject: [PATCH 03/67] Fix misc-macro-parentheses warnings in Gallery2.

Add parentheses around negative literals and parameters beside operators.
Bug: 28705665
Test: build with clang-tidy

Change-Id: Icdb2d003e77f6b0dfac7d9c30dec0ee80c13e3cd
---
 jni/filters/filters.h            | 12 ++++++------
 jni_jpegstream/src/error_codes.h |  8 ++++----
 2 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/jni/filters/filters.h b/jni/filters/filters.h
index 55c260f06..a46b3cedc 100644
--- a/jni/filters/filters.h
+++ b/jni/filters/filters.h
@@ -25,23 +25,23 @@
 
 typedef unsigned int Color;
 
-#define SetColor(a, r, g, b) ((a << 24) | (b << 16) | (g << 8) | (r << 0));
+#define SetColor(a, r, g, b) (((a) << 24) | ((b) << 16) | ((g) << 8) | ((r) << 0));
 #define GetA(color) (((color) >> 24) & 0xFF)
 #define GetB(color) (((color) >> 16) & 0xFF)
 #define GetG(color) (((color) >> 8) & 0xFF)
 #define GetR(color) (((color) >> 0) & 0xFF)
 
-#define MIN(a, b) (a < b ? a : b)
-#define MAX(a, b) (a > b ? a : b)
+#define MIN(a, b) ((a) < (b) ? (a) : (b))
+#define MAX(a, b) ((a) > (b) ? (a) : (b))
 
 #define LOG(msg...) __android_log_print(ANDROID_LOG_VERBOSE, "NativeFilters", msg)
 
 #define JNIFUNCF(cls, name, vars...) Java_com_android_gallery3d_filtershow_filters_ ## cls ## _ ## name(JNIEnv* env, jobject obj, vars)
 
 #define RED i
-#define GREEN i+1
-#define BLUE i+2
-#define ALPHA i+3
+#define GREEN (i+1)
+#define BLUE (i+2)
+#define ALPHA (i+3)
 #define CLAMP(c) (MAX(0, MIN(255, c)))
 
 __inline__ unsigned char  clamp(int c)
diff --git a/jni_jpegstream/src/error_codes.h b/jni_jpegstream/src/error_codes.h
index be55a0099..f224b4a09 100644
--- a/jni_jpegstream/src/error_codes.h
+++ b/jni_jpegstream/src/error_codes.h
@@ -17,10 +17,10 @@
 #ifndef JPEG_ERROR_CODES_H_
 #define JPEG_ERROR_CODES_H_
 
-#define J_DONE                    -4
-#define J_EXCEPTION               -3
-#define J_ERROR_BAD_ARGS          -2
-#define J_ERROR_FATAL             -1
+#define J_DONE                    (-4)
+#define J_EXCEPTION               (-3)
+#define J_ERROR_BAD_ARGS          (-2)
+#define J_ERROR_FATAL             (-1)
 #define J_SUCCESS                 0
 
 #endif // JPEG_ERROR_CODES_H_
-- 
2.17.1

